exporter/collector/googlemanagedprometheus/naming.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlemanagedprometheus
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func GetMetricName(baseName string, metric pdata.Metric) string {
+	return baseName + gmpMetricSuffix(metric)
+}
+
+func gmpMetricSuffix(metric pdata.Metric) string {
+	switch metric.DataType() {
+	case pmetric.MetricDataTypeSum:
+		return "/counter"
+	case pmetric.MetricDataTypeGauge:
+		return "/gauge"
+	case pmetric.MetricDataTypeSummary:
+		return "/summary"
+	case pmetric.MetricDataTypeHistogram:
+		return "/histogram"
+	default:
```

Comment:
Should this return empty string or an error? What other scenarios can fall through to here?

Reply:
The only case that this handled was exponential histograms.  I've added a specific case for it.  What suffix should we put on exponential histograms?

Reply:
Can exponential histograms be modeled the same way as standard Prometheus histograms? GMP doesn't specifically know about exponential histograms (since it's not something Prometheus knows about either) so if it requires any special handling, it won't be supported and we should return an error.

Reply:
When I implement normalization, i'll drop exponential histograms.  Then it won't be possible for those to fall through to here.  Should I just remove that case then?

Reply:
Yes, let's remove the case. I'd still suggest returning an error in the default case to be cautious.

Reply:
Removed and it now returns an error.